File: maxhanna.client/src/services/datacontracts/nexus/nexus-battle-outcome.ts
import { User } from "../user/user";

export interface NexusBattleOutcome {
  battleId: number;
  originUser: User;
  originCoordsX: number;
  originCoordsY: number;
  destinationUser: User;
  destinationCoordsX: number;
  destinationCoordsY: number;
  timestamp: string;
  attackingUnits: Record<string, number>;
  defendingUnits: Record<string, number>;
  attackingLosses: Record<string, number>;
  defendingLosses: Record<string, number>;
  defenderUnitsNotInVillage: Record<string, number>;
  defenderBuildingLevels: Record<string, number>;
  defenderGold: number;
  defenderGoldStolen: number; 
}

File: maxhanna.Server/Controllers/DataContracts/Nexus/NexusBattleOutcome.cs
using maxhanna.Server.Controllers.DataContracts.Users;

namespace maxhanna.Server.Controllers.DataContracts.Nexus
{

	public class NexusBattleOutcome
	{
		public int BattleId { get; set; }
		public User? OriginUser { get; set; }
		public int OriginCoordsX { get; set; }
		public int OriginCoordsY { get; set; }
		public User? DestinationUser { get; set; }
		public int DestinationCoordsX { get; set; }
		public int DestinationCoordsY { get; set; }
		public DateTime Timestamp { get; set; }
		public Dictionary<string, int?>? AttackingUnits { get; set; }
		public Dictionary<string, int?>? DefendingUnits { get; set; }
		public Dictionary<string, int?>? AttackingLosses { get; set; }
		public Dictionary<string, int?>? DefendingLosses { get; set; }
		public Dictionary<string, int?>? DefenderBuildingLevels { get; set; }
		public Dictionary<string, int?>? DefenderUnitsNotInVillage { get; set; }
		public Decimal? DefenderGold { get; set; }
		public Decimal DefenderGoldStolen { get; set; }
	}
}